Assets/Scripts/Features/Gameplay/Camera/CameraDragging.cs

```diff
@@ -0,0 +1,202 @@
+﻿using UnityEngine;
+using CM.Events;
+
+/// <summary>
+/// Controls the 2 dimensional movement of the Camera.
+/// </summary>
+public class CameraDragging : MonoBehaviour
+{
+    /// <summary>
+    /// A simple event to call when the camera is being moved.
+    /// </summary>
+    public event SimpleEvent OnDragging;
+
+    private Camera _camera;
+
+    private Vector3 _previousPosition;
+    private Vector3 _touchOrigin;
+
+    private bool _isDragging;
+
+    [Tooltip("The boundaries of the Camera movement")]
+    [SerializeField]
+    private Boundary2D _boundary;
+
+    [Tooltip("The speed of the camera dragging, also affects movement by keyboard.")]
+    [SerializeField]
+    private float _draggingSpeedMultiplier = 1;
+
+    [Tooltip("The threshold for the camera dragging. If the mouse movement is bigger than the threshold, begin dragging.")]
+    [SerializeField]
+    private float _draggingThreshold = 0.001f;
+
+    [SerializeField]
+    private bool _canDrag = false;
+
+    private void Start()
+    {
+        // Set _camera to the Main Camera.
+        _camera = Camera.main;
+    }
+
+    private void Update()
+    {
+        if (!_canDrag)
+            return;
+
+#if UNITY_EDITOR || UNITY_STANDALONE
+
+        // Movement via Dragging
+        if (Input.GetMouseButtonDown(0))
+            SaveOriginPositions();
+
+        if (Input.GetMouseButton(0))
+            MoveCamera();
+
+        if (Input.GetMouseButtonUp(0))
+        {
+            _previousPosition = transform.position;
+            _touchOrigin = _camera.ScreenToViewportPoint(Input.mousePosition);
+            _isDragging = false;
+        }
+
+        // Movement via Keyboard
+        Vector3 newPosition = new Vector3(0, 0, 0);
+        if (Input.GetAxisRaw("Vertical") != 0)
+            newPosition.z += Input.GetAxisRaw("Vertical")/5 * _draggingSpeedMultiplier;
+
+        if (Input.GetAxisRaw("Horizontal") != 0)
+            newPosition.x += Input.GetAxisRaw("Horizontal")/5 * _draggingSpeedMultiplier;
+
+
+        transform.position += newPosition;
+
+        ClampToBounds();
+
+#endif
+
+    }
+
+    private void SaveOriginPositions()
+    {
+        _previousPosition = transform.position;
+
+#if UNITY_EDITOR || UNITY_STANDALONE
+
+        _touchOrigin = _camera.ScreenToViewportPoint(Input.mousePosition);
+
+#endif
```

Comment:
All #if and #endif statements can be removed. This was only needed for the previous project because we needed this to work on android, while testing on pc with other controls.